flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -410,9 +424,11 @@ private void seizeCapacity() {
      * (once the record is not blocked).
      *
      * @param callback the callback to run if it finishes (once the record is not blocked).
+     * @param overdraft whether to overdraft the in-flight buffer.
      */
-    public StateFuture<Void> syncPointRequestWithCallback(ThrowingRunnable<Exception> callback) {
-        return handleRequest(null, StateRequestType.SYNC_POINT, null)
+    public StateFuture<Void> syncPointRequestWithCallback(
+            ThrowingRunnable<Exception> callback, boolean overdraft) {
```

Comment:
nit: overdraught -> allowOverdraught

Reply:
I'd name it `allowOverdraft` everywhere. It seems 'overdraught' and 'overdraft' are same but 'overdraft' is more common in github (search the two words above).